lib/collection/src/operations/verification/update.rs

```diff
@@ -252,3 +264,198 @@ fn check_collection_payload_size_limit(
 
     Ok(())
 }
+
+/// Compute a non-empty mapping of multivector limits by name.
+///
+/// Uses a tiny map as we expect a small number of multivectors to be configured per collection in strict mode.
+///
+/// Return None if no multivectors are configured with strict mode
+async fn multivector_limits_by_name(
+    collection: &Collection,
+    multivector_strict_config: &StrictModeMultivectorConfig,
+) -> Option<TinyMap<String, usize>> {
+    // If no multivectors strict mode no need to check anything.
+    if multivector_strict_config.config.is_empty() {
+        return None;
+    }
+    let collection_guard = collection.collection_config.read().await;
+
+    let multivector_max_size_by_name: TinyMap<String, usize> = collection_guard
+        .params
+        .vectors
+        .params_iter()
+        .filter_map(|(name, config)| config.multivector_config.map(|_strict_config| name)) // keep only configured multivectors
+        .filter_map(|multi_vector_name| {
+            multivector_strict_config
+                .config
+                .get(multi_vector_name)
+                .and_then(|config| config.max_vectors)
+                .map(|max_vectors| (multi_vector_name, max_vectors))
+        })
+        .map(|(name, max_vectors)| (name.to_string(), max_vectors))
+        .collect();
+    drop(collection_guard);
```

Comment:
This seems a bit complicated to me.

Can't we just clone `multivector_strict_config.config` and [`retain`](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.retain) only the known vector names? Or is there more to it that I'm not immediately seeing?

Reply:
Thank you for paying attention :+1: 

I was able to simplify the implementation, TBH I do not remember why I fetched the collection config again given that it is the same value provided as input .